x-pack/elastic-agent/pkg/core/monitoring/server/process.go

```diff
@@ -67,8 +67,14 @@ func processHandler(statsHandler func(http.ResponseWriter, *http.Request) error)
 			// proxy stats for elastic agent process
 			return statsHandler(w, r)
 		}
+		// TODO: allowlist of accepted endpoints to proxy to?
+		beatsEndpoint := vars["beatsEndpoint"]
 
-		metricsBytes, statusCode, metricsErr := processMetrics(r.Context(), id)
+		endpoint, err := generateEndpoint(id)
```

Comment:
I pulled this out so I could inject my own endpoint into `processMetrics`, to make testing easier.